pkg/ns/ns.go

```diff
@@ -179,7 +191,28 @@ func (ns *netNS) Close() error {
 	return nil
 }
 
+func (ns *netNS) validate(action string) error {
+	switch {
+	case ns.file == nil:
+		return fmt.Errorf(
+			"%q failed: no filedescriptor associated",
+			action,
+		)
+	case ns.mounted == false:
+		return fmt.Errorf(
+			"%q failed: no network namespace mounted on file %q",
+			action, ns.file.Name,
+		)
+	}
+
+	return nil
+}
+
 func (ns *netNS) Do(toRun func(NetNS) error) error {
+	if err := ns.validate("Do"); err != nil {
```

Comment:
Why bother calling validate() here?  Shouldn't ensuring validity on GetNS() be sufficient?  If we hold an open file handle to the namespace, it can't have been closed yet and it can't have been unmounted, so it should be sufficient to check that it's NSFS_MAGIC on GetNS() and then we're good, I'm pretty sure.